hazelcast/proxy/vector_collection.py

```diff
@@ -80,6 +168,25 @@ def put_all(self, map: Dict[Any, Document]) -> Future[None]:
         return combine_futures(futures)
 
     def put_if_absent(self, key: Any, document: Document) -> Future[Optional[Document]]:
+        """Associates the specified key with the given Document if it is not
+        already associated.
+
+        Warning:
+            This method returns a clone of the previous Document, not the original
+            (identically equal) Document previously put into the VectorCollection.
+
+        Warning:
+            This method uses ``__hash__`` and ``__eq__`` methods of binary form
+            of the key, not the actual implementations of ``__hash__`` and
+            ``__eq__`` defined in key's class.
```

Comment:
What does this mean? Isn't this true for all objects for all APIs of the Hazelcast client? We don't specify this for other APIs though. Can it be confusing here as extra?

Reply:
Took it straight from `Map.put_if_absent`.

Reply:
`put_if_absent` checks for equality of the serialized binary object on the server side. Not all methods do that.